src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -47,10 +47,7 @@ abstract class BaseNode implements NodeInterface
      */
     public function __construct(?string $name, NodeInterface $parent = null, string $pathSeparator = self::DEFAULT_PATH_SEPARATOR)
     {
-        if (null === $name) {
-            $name = '';
-        }
-        if (false !== strpos($name, $pathSeparator)) {
+        if (false !== strpos($name = (string) $name, $pathSeparator)) {
```

Comment:
Changing a variable, esp it's type within a call and later rely on this is really not-obivous. It took me a while to see this.

Reply:
I understand, but this reduces merge conflicts as you know.